### ISSUE TITLE
Add documentation for "odo link" using a java and database example

### ISSUE THREAD

**Issue by dharmit** [openshift/odo#4116]
## What mistake did you find / what is missing in the documentation?
`odo link` helps link an odo component to an Operator backed service. Currently we have an example of nodejs + etcd cluster on odo.dev. We need to add a java example where the component connects with a database started from an Operator.

## What is the relevance of it?
This is relevant because odo supports java but we don't have a document to explain linking of a java component with an Operator backed service.

/kind documentation
/area service-operators
/area linking
/assign

**Comment by dharmit** [comment]
cc @burrsutter

**Comment by dharmit** [comment]
Tasks that need to be done here:
1. Modify the code https://github.com/burrsutter/spring-petclinic/ to connect to a DB service backed by an Operator available from operatorhub.io - either Postgres or MariaDB
2. Add documentation that can be published on odo.dev

**Comment by dharmit** [comment]
Moving this to "For Consideration" since @kadel is working on making a demo around this and, while he doesn't expect to have a fully working demo, we expect more issues related to docs or "odo link" or "odo service" as a result of his effort.

**Comment by boczkowska** [comment]
@kadel @dharmit do I understand correctly that this one is blocked by https://github.com/openshift/odo/pull/4413 since most operators use Helm? 
I've tried to test it myself and I don't think I made it work with any of the operators in the Operator Catalogue now. =\ Is there a way around it?

There's a draft PR for it https://github.com/openshift/openshift-docs/pull/28738 with placeholder values for ope feel free to leave comments here or in the PR itself.

**Comment by kadel** [comment]
> @kadel @dharmit do I understand correctly that this one is blocked by #4413 since most operators use Helm?

Nope, this is not blocked. Helm is just another way to package applications on Kubernetes. It is unrelated to operators and this issue. 

> I've tried to test it myself and I don't think I made it work with any of the operators in the Operator Catalogue now. =\ Is there a way around it?

The experience with the Operator catalog is broken in odo right now :-/ Not broken in the sense that it is not working. There are prerequisites that odo doesn't docuemnt.

**Comment by burrsutter** [comment]
We have moved the demo to this location as it has become one of the primary ways we show off "developer experience" https://github.com/redhat-developer-demos/spring-petclinic

**Comment by boczkowska** [comment]
> > @kadel @dharmit do I understand correctly that this one is blocked by #4413 since most operators use Helm?
> 
> Nope, this is not blocked. Helm is just another way to package applications on Kubernetes. It is unrelated to operators and this issue.
> 
> > I've tried to test it myself and I don't think I made it work with any of the operators in the Operator Catalogue now. =\ Is there a way around it?
> 
> The experience with the Operator catalog is broken in odo right now :-/ Not broken in the sense that it is not working. There are prerequisites that odo is not documenting anywhere.

Shall we document these prerequisites or leave the docs till the procedure is "smooth"? 
From what I understand this is one of the most common user-stories for odo users, even documentation with "workarounds" can be helpful.

**Comment by kadel** [comment]
@boczkowska as you mention this is one of the most common user-stories. And especially because it is currently not as "smooth" as we would like we need to have good documentation including all necessary workarounds.

**Comment by dharmit** [comment]
@kadel @boczkowska I made some changes to a doc that was being prepared by @ajm01 for Java OpenLiberty + PostgreSQL. Would it make sense to add this to odo.dev (minikube based example) as an example of Java+DB? [Link to doc](https://github.com/dharmit/application-stack-samples/blob/minikube-readme/jpa/README-minikube.md).

I can make changes to make it more suitable for odo.dev site.

**Comment by boczkowska** [comment]
@dharmit thanks, this looks like it can even be used for downstream documentation too (obv, for OpenShift clusters), I'll test it today, and if all works as expected and update my PR in OCP repo accordingly. 
I can create a PR for upstream/odo.dev also, while I'm at it.

**Comment by dharmit** [comment]
> this looks like it can even be used for downstream documentation too (obv, for OpenShift clusters), I'll test it today, and if all works as expected and update my PR in OCP repo accordingly.

There's a [separate doc for OCP](https://github.com/ajm01/application-stack-samples/blob/minikube-readme/jpa/README.md) on the repo I cloned from. 

>  I can create a PR for upstream/odo.dev also, while I'm at it.

That would be very helpful. However, on odo.dev, we usually to put up steps that can be performed on minikube. The doc I linked earlier in https://github.com/openshift/odo/issues/4116#issuecomment-783355802 can be used for it. 

Let me know if it's fine by you to add doc based on minikube for odo.dev.
